nistats/glm.py

```diff
@@ -58,183 +47,394 @@ def data_scaling(Y):
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    t_r: float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    standardize : boolean, optional
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    percent_signal_change: bool, optional,
+        If True, fMRI signals is scaled to percent of the mean value
+        Incompatible with standardize (overrides standardize).
+
+    n_jobs : integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    verbose : integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+
+    noise_model : {'ar1', 'ols'}, optional
+        the temporal variance model. Defaults to 'ar1'
+
+    Attributes
+    ----------
+    labels : array of shape (n_voxels),
+        a map of values on voxels used to identify the correseponding the model
+
+    results : dictionary,
+        keys correspond to the different labels values
+        values are RegressionResults instances corresponding to the voxels
+
+    
+    """
 
-    def get_beta(self, column_index=None):
-        """Acessor for the best linear unbiased estimated of model parameters
+    def __init__(self, mask=None, target_affine=None, target_shape=None,
+             low_pass=None, high_pass=None, t_r=None, smoothing_fwhm=None,
+             memory=Memory(None), memory_level=1, standardize=False,
+             percent_signal_change=True, verbose=1, n_jobs=1,
+             noise_model='ar1'):
+        self.mask = mask
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.standardize = standardize
+        self.n_jobs = n_jobs
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.smoothing_fwhm = smoothing_fwhm
+        self.noise_model = noise_model
+        self.percent_signal_change = percent_signal_change
+        if self.percent_signal_change:
+            self.standardize = False
+
+    def fit(self, imgs, design_matrices):
+        """ Note: design_matrices is the design matrix !
```

Comment:
Should this go in a note section? At the very least, there is a missing short description.